File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        let sum = 1;
        let numberOfZero = 0
        for(const num of nums){
            if(num == 0){
                numberOfZero +=1
            }
            else{
                sum *=num
            }
        }
        if(numberOfZero >1){
            return Array.from({length: nums.length}, ()=> 0)
        }
        for(let i = 0; i< nums.length; i++){
            if(numberOfZero == 0){
                nums[i] = sum/nums[i]
            }
            else{
                if(nums[i]==0){
                    nums[i]= sum
                }
                else{
                    nums[i] = 0
                }
            }
        }
        return nums
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-1.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        l = 0
        nums.sort()
        res = []
        for i in range (len(nums)-2):
            if(i>0 and nums[i]==nums[i-1]):
                continue
            
            if(nums[i]>0):
                break
            
            l,r = i+1, len(nums)-1

            while(l<r):
                sum = nums[i] + nums[l] + nums[r]
                if(sum == 0 ):
                    res.append([nums[i], nums[l], nums[r]])
                    l+=1
                    r-=1
                    while(l<r and nums[l] == nums[l-1]):
                        l+=1
                    while (l<r and nums[r]==nums[r+1]):
                        r-=1
                    
                elif(sum<0):
                    l+=1
                else:
                    r-=1
            
        return res